hyperapp.d.ts

```diff
@@ -0,0 +1,212 @@
+    | ActionWithEffects<State, DPayload, DPayload>  // (state, data) => ({ ... })  | (state, data) => ([{ ... }, effect1, ...])
+);
+
+/** Usable to 1st argument of `dispatch`, make strict for `init` (state and default payload are always undefined)
+ *
+ * @memberOf [App]
+ */
+export type DispatchableOnInit<State, CPayload> = (
+    State
+    | ([(ActionOnInit<State, CPayload>), PayloadCreator<undefined, CPayload>])
+    | ([(ActionOnInit<State, CPayload>), CPayload])
+    | ActionOnInit<State, undefined>
+);
+
+/** A definition of `dispatch`. This is passed as an argument of effect or subscription runner.
+ * 
+ * @memberOf [App]
+ */
+export type Dispatch<State, NextPayload> = (obj: Dispatchable<State, NextPayload>, data: NextPayload) => void;
+
+
+/** An effect runner. It is actually invoked when effect is reflected.
+ * 
+ * @memberOf [App]
+ */
+export type EffectFunc<State, Props, AcceptDPayload = any, NextPayload = void> = (dispatch: Dispatch<State, NextPayload>, props: Props, payload: AcceptDPayload) => void;
+
+/** An effect as the result of an action.
+ * 
+ * @memberOf [App]
+ */
+export type Effect<State = any, AcceptDPayload = any> = [EffectFunc<State, any, AcceptDPayload>, any];
+
+/** An subscription runner. It is actually invoked when effect is reflected.
+ * 
+ * @memberOf [App]
+ */
+export type SubscriptionFunc<State, Props, NextPayload> = (dispatch: Dispatch<State, NextPayload>, props: Props) => (() => void);
+
+/** A reference to an subscription to be managed by Hyperapp, with optional additional parameters
+ * 
+ * @memberOf [App]
+ */
+export type Subscription<State = any> = [SubscriptionFunc<State, any, any>, any];
+
+/** The result of an action.
+ *
+ * @memberOf [App]
+ */
+export type ActionResultWithEffects<State, AcceptDPayload> = State | [State, ...Effect<State, AcceptDPayload>[]];
+
+/** The interface for a single action implementation. (without effect)
+ *
+ * @memberOf [App]
+ */
+export type Action<State, Payload = void> = (state: State, data: Payload) => State;
+
+/** The interface for a single action implementation.
+ *
+ * @memberOf [App]
+ */
+export type ActionWithEffects<State, Payload, AcceptDPayload> = (state: State, data: Payload) => ActionResultWithEffects<State, AcceptDPayload>;
+
+/** The interface for a single action implementation, make strict for `init` (given state are always undefined)
+ *
+ * @memberOf [App]
+ */
+export type ActionOnInit<State, Payload = void> = (state: undefined, data: Payload) => ActionResultWithEffects<State, any>;
+
+/** The view function describes the application UI as a tree of VNodes.
+ * @returns A VNode tree.
+ * @memberOf [App]
+ */
+export interface View<State> {
+    (state: State): VNode<object> | null;
+}
+
+/** The possible response types for the subscription callback for an application
+ * 
+ * @memberOf [App]
+ */
+export type SubscriptionsResult<State> = | (Subscription<State> | boolean)[] | Subscription<State>;
```

Comment:
I think `subscriptions` can returns a boolean without array.
Currently, the following code is compilable, but it is best to specify it in the type definition.

```tsx
subscriptions: state => state.enabled && subs()
```